x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/calendars/ScheduledEventTests.java

```diff
@@ -119,4 +120,8 @@ public void testLenientParser() throws IOException {
             ScheduledEvent.LENIENT_PARSER.apply(parser, null);
         }
     }
+
+    private static ZonedDateTime nowWithMillisResolution() {
```

Comment:
Should we factor this out into a base class now it's used in two places (the other is `GetWatchResponseTests`)?  `ESTestCase`?  What do you think @spinscale?

Reply:
this is very tempting as watcher/sql is already doing the same (in a slightly different way) 
https://github.com/elastic/elasticsearch/pull/36877